.github/workflows/ci.yml

```diff
@@ -78,6 +78,18 @@ jobs:
     steps:
     - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
 
+    - uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
```

Comment:
Is it necessary to cache these in order for `cargo metadata` to work?

Reply:
I'm utilizing the provided [example](https://github.com/actions/cache/blob/main/examples.md#rust---cargo) directly. I'm uncertain about whether the command `cargo metadata` will be used or not. As far as I understand, this command checks if the package has not been installed yet. If the package has been installed, it will be stored in the `~/.cargo/registry` directory. Given this, it seems plausible that we could potentially cache only the `~/.cargo/registry/index` and the `~/.cargo/registry/cache` directory.

Reply:
I've tested this [here](https://github.com/zoo868e/zerocopy/actions/runs/6042393394/job/16397422883), and it appears that only caching the `~/.cargo/registry/index` is not sufficient for the `cargo metadata` command.

Reply:
Looks like the latest version of the PR only includes `~/.cargo/registry/index/` and `~/.cargo/registry/cache`; I assume your testing revealed that just caching these two is sufficient?

Reply:
Yes, I tested it on my forked repository, and it only requires these two directories to prevent downloading the crates every time.

Reply:
Sounds good!